src/poetry/repositories/pool.py

```diff
@@ -1,59 +1,77 @@
 from __future__ import annotations
 
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import IntEnum
+from enum import auto
 from typing import TYPE_CHECKING
 
+from poetry.repositories.abstract_repository import AbstractRepository
 from poetry.repositories.exceptions import PackageNotFound
-from poetry.repositories.repository import Repository
 
 
 if TYPE_CHECKING:
     from poetry.core.constraints.version import Version
     from poetry.core.packages.dependency import Dependency
     from poetry.core.packages.package import Package
 
+    from poetry.repositories.repository import Repository
 
-class Pool(Repository):
+
+class RepositoryPriority(IntEnum):
+    # The order of the members below dictates the actual priority. The first member has
+    # top priority.
+    DEFAULT = auto()
+    PRIMARY = auto()
+    SECONDARY = auto()
+
+
+@dataclass
+class PrioritizedRepository:
+    repository: Repository
+    priority: RepositoryPriority
+
+
+class Pool(AbstractRepository):
```

Comment:
It looks like the name got reverted to `Pool`. Sorry if I was unclear before -- I'd like to keep the name `RepositoryPool` in this PR, even if we do the file-level renames in another.

Reply:
That will still break [the plugin test code](https://github.com/python-poetry/poetry-plugin-export/blob/main/tests/conftest.py#L16), so I think it is best to combine the file+ object renaming in a single PR. Do you agree?

Reply:
Ah, I missed that -- fair enough!